include/zephyr/drivers/comparator.h

```diff
@@ -70,10 +70,9 @@ __syscall int comparator_get_output(const struct device *dev);
 
 static inline int z_impl_comparator_get_output(const struct device *dev)
 {
-	const struct comparator_driver_api *api =
-		(const struct comparator_driver_api *)dev->api;
+	__ASSERT_NO_MSG(DEVICE_API_IS(comparator, dev));
```

Comment:
Offline discussion resulted in leaving out the asserts. The reason being that encapsulates APIs (eg I3c has i2c) can be used downstream.

Reply:
Removed asserts